strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactControllerTest.java

```diff
@@ -125,4 +127,56 @@ public void testPackageCommonFlow(@NpmRepository(repositoryId = REPOSITORY_RELEA
                .header(HttpHeaders.CONTENT_LENGTH, equalTo(String.valueOf(Files.size(packagePath))));
     }
 
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    public void addUserTest(@NpmRepository(repositoryId = REPOSITORY_RELEASES)
+                                    Repository repository)
+    {
+        String url = getContextBaseUrl() + "/storages/{storageId}/{repositoryId}/" + NpmLayoutProvider.NPM_USER_PATH + "{username}";
+
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        NpmUser strongboxUser1 = new NpmUser("admin", "password");
+        NpmUser strongboxUser2 = new NpmUser("deployer", "password");
+        NpmUser nonStrongboxUser = new NpmUser("notARealUser", "notARealPassword");
+
+        //can login with strongbox user
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(strongboxUser1)
+               .when()
+               .put(url, storageId, repositoryId, strongboxUser1.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.CREATED.value());
+
+        //can login with another strongbox user after login
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(strongboxUser2)
+               .when()
+               .put(url, storageId, repositoryId, strongboxUser2.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.CREATED.value());
+
+        //can't login when the url username differs from the body
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(strongboxUser1)
+               .when()
+               .put(url, storageId, repositoryId, nonStrongboxUser.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.UNAUTHORIZED.value());
+
+        //can't login with non-strongbox user
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(nonStrongboxUser)
+               .when()
+               .put(url, storageId, repositoryId, nonStrongboxUser.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.UNAUTHORIZED.value());
```

Comment:
@LeifBrooks can you please add the following case
```
//can't login with `Authorization: Basic <strongbox-user-credentials>` and non-strongbox user login request body
```

Reply:
You got it

Reply:
I've added the positive of this test as well. Can log in with basic auth and strongbox user

Reply:
done